tutorials/Land/Soil/PhaseChange/freezing_front.jl

```diff
@@ -0,0 +1,478 @@
+# water fraction approaches a constant around 0.075 in the frozen region, rather than the residual fraction
+# of 0.019, or 0. This behavior is observed, for example, in the experiments of [5].
+
+# Although this approach may indicate that we should replace the pressure head appearing in the
+# diffusive water flux term in Richards equation [3], we do not do so at present. As such, we may not be modeling
+# the flow of water around the freezing front properly. However, we still observe cryosuction, which
+# is the flow of water towards the freezing front, from the unfrozen side. As the water freezes, the liquid
+# water content drops,
+# setting up a larger gradient in matric potential across the freezing front, which generates upward flow
+# against gravity. This is evident because the total water content at the top is larger at the end of the
+# simulation
+# than it was at `t=0` (when it was 0.33).
+
+# This model differs from others [2,3,6] in that it requires us to set a timescale for the phase change, `τ`.
+# In a first-order
+# phase transition, the temperature is fixed while the necessary latent heat is either lost or gained by the
+# system.  Ignoring
+# changes in internal energy due to flowing water, we would expect
+
+
+# ``
+# \partial_t ρe_{int} \approx (ρ_l c_l \partial_t θ_l + ρ_i c_i \partial_t θ_i) (T-T_0) -ρ_i L_f \partial_t θ_i
+# ``
+
+# ``
+# = [(c_i-c_l) (T-T_0) -L_f]F_T \approx -L_f F_T
+# ``
+
+# or
+
+# ``
+# F_T ∼ \frac{κ|∇²T|}{L_f} ∼\frac{κ}{c̃ Δz²}\frac{c̃ |∂zT| Δz}{L_f}
+# ``
+
+# suggesting
+
+# ``
+# τ ∼ τ_{LTE}\frac{ρ_lL_f (ν-θ_r)}{c̃ |∂zT| Δz}
+# ``
+
+# with
+
+# ``
+# τ_{LTE}= c̃ Δz²/κ
+# ``
+
+# This is the value we use. This seems to work adequately for modeling freezing front propagation and
+# cryosuction, via comparisons with [1], but we plan to revisit it in the future. For example,
+# we do not see a strong temperature plateau at the freezing point [e.g. 5],
+# which we would expect while the phase change is occuring.
+
+# # References
+# [1] Mizoguchi, M. (1990)
+# Water, heat and salt transport in freezing soil. Ph.D. thesis; University of Tokyo.
+
+# [2] Hansson, K. et al. (2004)
+# Water Flow and Heat Transport in Frozen Soil: Numerical Solution and Freeze–Thaw Applications
+# Vadose Zone Journal 3:693-704
+
+# [3] Dall'Amico, M. (2011) A robust and energy-conserving model of freezing variably-saturated soil.
+# The Cryosphere, 5, 469–484
+
+# [4] Kurylyk, B. and Watanabe, K. (2013)
+# Review: The mathematical representation of freezing and thawing processes in variably-saturated,
+# non-deformable soils. Advances in Water Resources, 60:160-177
+
+# [5] Watanabe, K. et al. (2011)
+# Freezing experiments on unsaturated sand, loam, and silt loam. Annals of Glaciology 52(58)
+
+# [6] Painter, Scott L. (2011)
+# Three-phase numerical model of water migration in partially frozen geological media:
+# model formulation, validation, and applications
+# Computational Geosci 15:69-85
```

Comment:
Should we add these as entries to the `docs/bibliography.bib`? That way references will be more compact per tutorial.

Reply:
yessss, we probably should :)

Reply:
should be good!